src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,164 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static final Object NULL_VALUE = null;
+    private static final String UNREAL_LOGIN1 = "nuefdd";
+    private static final String UNREAL_LOGIN2 = "dfghjkl";
+    private static final String LOGIN = "Login";
+    private static final String PASSWORD = "password";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final String LOGIN1 = "login1";
+    private static final String LOGIN2 = "login2";
+    private static final String LOGIN3 = "login3";
+    private static final String PASSWORD1 = "password1";
+    private static final String PASSWORD2 = "password2";
+    private static final String PASSWORD3 = "password3";
+    private static final int AGE1 = 22;
+    private static final int AGE2 = 40;
+    private static final int AGE3 = 18;
+    private static final int SMALL_AGE = 12;
+    private static final int UNREAL_AGE = -10;
+    private static final long USER_ID1 = 1L;
+    private static final long USER_ID2 = 2L;
+    private static final long USER_ID3 = 3L;
+    private static RegistrationService registrationService;
+    private static StorageDao storage;
+    private User user1;
+    private User user2;
+    private User user3;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storage = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user1 = new User(USER_ID1,LOGIN1, PASSWORD1, AGE1);
+        user2 = new User(USER_ID2,LOGIN2, PASSWORD2, AGE2);
+        user3 = new User(USER_ID3,LOGIN3, PASSWORD3, AGE3);
```

Comment:
no need for 3 users, 1 default valid user is enough, and you can just modify some of its fields up to you needs in each test case so that all other user properties would be valid and only the one that you are testing is not (in case it is `notOk` case of course)